CMakeLists.txt

```diff
@@ -73,45 +79,64 @@ if (is_multi_config)
 endif()
 
 function(apply_output_rules target)
-  set_property(TARGET ${target} PROPERTY "ARCHIVE_OUTPUT_DIRECTORY" "${CMAKE_BINARY_DIR}/${in_build_prefix}lib")
-  set_property(TARGET ${target} PROPERTY "LIBRARY_OUTPUT_DIRECTORY" "${CMAKE_BINARY_DIR}/${in_build_prefix}lib")
-  set_property(TARGET ${target} PROPERTY "RUNTIME_OUTPUT_DIRECTORY" "${CMAKE_BINARY_DIR}/${in_build_prefix}bin")
+  cmake_parse_arguments(PARSE_ARGV 1 OUTPUT_RULES "" "TARGET_DIR" "")
+  set(output_prefix "${PROJECT_BINARY_DIR}/${in_build_prefix}")
+  set_property(TARGET ${target} PROPERTY "ARCHIVE_OUTPUT_DIRECTORY" "${output_prefix}lib/${OUTPUT_RULES_TARGET_DIR}")
+  set_property(TARGET ${target} PROPERTY "LIBRARY_OUTPUT_DIRECTORY" "${output_prefix}lib/${OUTPUT_RULES_TARGET_DIR}")
+  set_property(TARGET ${target} PROPERTY "RUNTIME_OUTPUT_DIRECTORY" "${output_prefix}bin/${OUTPUT_RULES_TARGET_DIR}")
 endfunction()
 
 include(cmake/jakt-executable.cmake)
 
 add_subdirectory(runtime)
 
-file(GLOB JAKT_STAGE0_SOURCES CONFIGURE_DEPENDS "bootstrap/stage0/*.cpp")
-file(GLOB JAKT_STAGE0_RUNTIME_SOURCES CONFIGURE_DEPENDS "bootstrap/stage0/runtime/*.cpp")
-file(GLOB JAKT_STAGE0_RUNTIME_LIB_SOURCES CONFIGURE_DEPENDS "bootstrap/stage0/runtime/*/*.cpp")
-list(APPEND JAKT_STAGE0_ALL_SOURCES
-    ${JAKT_STAGE0_SOURCES}
-    ${JAKT_STAGE0_RUNTIME_SOURCES}
-    ${JAKT_STAGE0_RUNTIME_LIB_SOURCES}
-)
+set(JAKT_TARGET_TRIPLE "${JAKT_TARGET_TARGET}" CACHE STRING "Target triple to use for cross-compilation (defaults to cmake's target triple)")
 
-# Note: This currently does not build under windows, but it is not needed by the runtime.
-list(FILTER JAKT_STAGE0_ALL_SOURCES EXCLUDE REGEX ".*AK/Time\.cpp$")
-list(FILTER JAKT_STAGE0_ALL_SOURCES EXCLUDE REGEX ".*AK/DOSPackedTime\.cpp$")
+if (NOT ${JAKT_TARGET_TRIPLE} STREQUAL ${JAKT_TARGET_TARGET})
```

Comment:
TARGET_TARGET and TARGET_TRIPLE? I know what this patch is supposed to do and I still don't understand the difference between these variables.

Is what's going on here that one of them is JAKT_HOST_TRIPLE and the other is JAKT_TARGET_TRIPLE?

We have access to CMAKE_HOST_SYSTEM_PROCESSOR and CMAKE_HOST_SYSTEM, are those what you're really looking for?

Reply:
I've probably named these in the worst way possible, here's what's going on:
- `JAKT_TARGET_TRIPLE` is the user-facing config thingy, defaults to `JAKT_TARGET_TARGET`
- `JAKT_TARGET_TARGET` is what CMake thinks the target triple is
- there's no `JAKT_HOST_TRIPLE` anymore.

Reply:
err...they both refer to the same thing, `TARGET_TRIPLE` is just there to let the user override it in case cmake is wrong
I can remove TARGET_TARGET and use TARGET_TRIPLE exclusively if that's what you prefer?

Reply:
Yeah that would be preferred lol. Just let the variable from the jakt-executable.cmake file be user-settable as a cache variable and it'll be great.